### PR TITLE
TST: Added tests to check if file exists for read_json.

### LINKED ISSUE #29102: Misleading error messages when opening inexistent json file
When opening a json file that doesn't exist with `read_json`, we get a `ValueError` with error messages like:

```python
>>> import pandas
>>> pandas.read_json('no_file.json')
ValueError: Unexpected character found when decoding 'null'

>>> pandas.read_json('a_dir/no_file.json')
ValueError: Expected object or value
```

It'd probably be better to raise an exception like:
```
FileNotFoundError: No such file: 'no_file.json'
```

### PR DESCRIPTION
**Pull request by cheungje**
- [x] closes #29102 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

#29102 already added code to fix the original issue of fixing misleading error messages when trying to open a nonexistent json file. This commit adds test cases.

### CODE REVIEW

File: pandas/tests/io/test_common.py [hunk 1]
```diff
@@ -265,7 +266,8 @@ def test_write_missing_parent_directory(self, method, module, error_class, fn_ex
             (pd.read_hdf, "tables", FileNotFoundError, "h5"),
             (pd.read_stata, "os", FileNotFoundError, "dta"),
             (pd.read_sas, "os", FileNotFoundError, "sas7bdat"),
-            (pd.read_json, "os", ValueError, "json"),
+            # (pd.read_json, "os", ValueError, "json"),
```
Comment (phofl) [inline pandas/tests/io/test_common.py:269]: Why did you remove this?

Which message is actually shown?

Comment (cheungje) [reply pandas/tests/io/test_common.py:269]: I removed this because the comments in #29102 asks to create test cases such that the errors returned by read_json returns a FileNotFoundError instead of a ValueError.

Comment (phofl) [reply pandas/tests/io/test_common.py:269]: You have to change the actual implementation to raise the correct error

**Comment by jreback** [comment]
@cheungje if you'd merge master and address comments

**Comment by jreback** [comment]
closing as stale
